From 25a7ad1bbe8d529ab6c5bb8557d39650d0b2378f Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?David=20K=C3=A5gedal?= <david@kagedal.org>
Date: Mon, 10 Feb 2014 22:00:15 +0100
Subject: [PATCH 3/4] Improve the library headers
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

This makes it possible to add to ELPA and other archives. Also this
adds rudimentary documentation explainint how to get started.

Signed-off-by: David K책gedal <david@kagedal.org>
Signed-off-by: Catalin Marinas <catalin.marinas@gmail.com>
---
 contrib/stgit.el | 31 +++++++++++++++++++++++++++++--
 1 file changed, 29 insertions(+), 2 deletions(-)

diff --git a/contrib/stgit.el b/contrib/stgit.el
index e25d433..7cd13c3 100644
--- a/contrib/stgit.el
+++ b/contrib/stgit.el
@@ -1,6 +1,30 @@
-;; stgit.el: An emacs mode for StGit
+;;; stgit.el --- major mode for StGit interaction
 ;;
-;; Copyright (C) 2007 David K책gedal <davidk@lysator.liu.se>
+;; Copyright (C) 2007-2013 David K책gedal
+;;
+;; Author: David K책gedal <davidk@lysator.liu.se>
+;; Homepage: http://www.procode.org/stgit
+;; Version: 0.17.1
+
+;;; Commentary:
+;;
+;; This is an interactive tool to interact with git branches using
+;; StGit.
+;;
+;; StGit is a command-line tool providing similar functionality to
+;; Quilt (i.e. pushing/popping patches to/from a stack) on top of Git.
+;; These operations are performed using Git commands and the patches
+;; are stored as Git commit objects, allowing easy merging of the
+;; StGit patches into other repositories using standard Git
+;; functionality.
+;;
+;; To start using the Emacs interface, run M-x stgit and select the
+;; git repository you are working in.
+;;
+;; To get quick help about the available keybindings in the buffer,
+;; press '?'
+
+;;; Installation:
 ;;
 ;; To install: put this file on the load-path and place the following
 ;; in your .emacs file:
@@ -9,6 +33,8 @@
 ;;
 ;; To start: `M-x stgit'
 
+;;; Code:
+
 (when (< emacs-major-version 22)
   (error "Emacs older than 22 is not supported by stgit.el"))
 
@@ -3057,3 +3083,4 @@ The initial setting is controlled by `stgit-default-show-committed'."
                           "Hiding"))))
 
 (provide 'stgit)
+;;; stgit.el ends here
-- 
1.9.1

